KIPs/kip-249.md

```diff
@@ -0,0 +1,375 @@
+	to_addr: str # The target address of backrun logic
+	call_gas_limit: bytes # The gas limit of backrun logic
+	call_data: bytes # The call data of backrun logic
+	bid_amount: bytes # The bid amount in Kei
+	from_signature: bytes # ECDSA.sign the hash typed data V4 with auction_type_hash based on EIP-712 by searcher's private key.
+	auc_signature: bytes # ECDSA.sign the from_signature based on EIP-191 by auctioneer's private key
+```
+
+The proposer verifies the \`BidDetails\` in the following order.
+
+1. Verify the both signatures from searcher and auctioneer.  
+   1. The auctioneer address will come from the AuctionEntryPoint contract.  
+2. Verify the \`Bid.blockNum \== currBlockNum \+ 1 && \!MiningNotStarted\`  
+   1. MiningNotStarted is a flag representing the current mining status.  
+3. Verify the \`Bid.bid\_amount \>= min\_bid\_amount\`.
+
+---
+
+### Block Building
+
+The verified bids will wait for the next block mining. When the mining starts, the proposer will place the bid results into block with following steps:
+
+1. Retrieve target tx hash from bids  
+2. Drop all txs that arrive after the early deadline except the target tx.  
+3. Drop all bids that have no corresponding target tx in the current tx list.  
+4. Verifying that the searcher has enough deposit balance to pay the \`bidAmount\` and \`gasFee\`. If not, drop all the bids from a searcher.  
+5. Verifying that the searcher has the correct nonce in AuctionEntryPoint.  
+6. Make an unsigned BidTx and bundle for each bid and place them after the target tx.
+
+#### BidTx
+
+The proposer shall inject a transaction denoted by BidTx, which calls the AuctionEntryPoint contract. The BidTx will follow the below rules:
+
+1. BidTx.type is 0x7802, which is the Ethereum Dynamic Fee transaction type.  
+2. BidTx.to is an AuctionEntryPoint contract.  
+3. BidTx.value is 0  
+4. BidTx.data shall be correctly encoded for \`AuctionEntyPoint.call\`.  
+5. BidTx.maxFeePerGas, BidTx.maxPriorityFeePerGas shall be the same as those of target tx.  
+6. BidTx.gasLimit shall be set large enough to cover the gas fee including \`Bid.call\_gas\_limit\`.
+
+After building the BidTx, the proposer will make a BidTx bundle based on the KIP-245.
+
+#### BidTx Bundling
+
+After making the BidTx, it will be bundled based on the KIP-245. Please note that the bid bundle doesn’t include target tx but single bid tx so that the target tx can’t be delayed/canceled due to bundling.
+
+```
+Original tx list: [Tx1, Tx2, Tx3, Tx4]
+Target tx: Tx2
+BidTx bundle: {targetTx: Tx2, bundleTxs: [B2]} // not [Tx2, B2]
+
+Final tx list: [Tx1, Tx2, [B2], Tx3, Tx4]
+```
+
+Please note that the proposer doesn’t sign the BidTx yet and it’ll be done right before executing the BidTx. This is because we can’t finalize the nonce of the proposer before executing all previous transactions and bundles when there're more than one bundle in the same block.  
+---
```

Comment:
```suggestion
Please note that the proposer doesn’t sign the BidTx yet and it’ll be done right before executing the BidTx. This is because we can’t finalize the nonce of the proposer before executing all previous transactions and bundles when there're more than one bundle in the same block.  

---
```